design/pen.behavior.md

```diff
@@ -0,0 +1,101 @@
+# Pen Behavior
+
+The pen behavior is a method of recording the path by which a user moves a grabbed object that they activate so that various scenearios 
+can be carried out that wish to capture the virtual drawing of a user.  This behavior will carry with it client side visuals
+to ensure low-latency feedback, but will supply the complete curve data needed by the MRE app to acto on once the drawing is done.
+The `PenBehavior` will be an extension of the `ToolBehavior`.
+
+## Architecture
+
+There will be a new `PenBehavior` that extends the `ToolBehavior` that will have the ability to generate a wet ink effect along the use 
+path on the client and provide the use path to the MRE app for generating a spline curve asset that could have a mesh to applied to it
+before sending it to all the clients to render. and a mesh to apply to this spline curve.
+
+### Behavior
+``` ts
+/**
+ * Pen behavior class containing the target behavior actions.
+ */
+export class PenBehavior extends ToolBehavior {
+	/** @inheritdoc */
+	public get behaviorType(): BehaviorType { return 'pen'; }
+
+	public onDrawEventReceived((drawData: Partial<DrawDataLike>[]) => void) {
+		// Register for draw events.
+	}
+
+	public offDrawEventReceived((drawData: Partial<DrawDataLike>[]) => void) {
+		// Unregister for draw events.
+	}
+```
+
+### Draw Data
+
+``` ts
+export interface DrawDataLike {
+	transform: Transform;
+	// Potentially additional data to come, such as:
+	pressure: number;
+}
```

Comment:
Are we assuming that each DrawData frame is spaced equally apart in time? The speed with which you draw might be important.

Reply:
We have discussed including time stamps in to the draw data.  The requirements there are still in flux.